### DISCUSSION TITLE
Ram issue

### DISCUSSION THREAD

**Discussion by ECEAthipanbose** [scalameta/metals-vscode#1460] [Q&A]
java -Dbloop.truncate-output-file-periodically=/home/local/USERECE/.local/share/scalacli/bloop/daemon/output -Xss4m -XX:MaxInlineLevel=20 -XX:+UseZGC -XX:ZUncommitDelay=30 -XX:ZCollectionInterval=5 -XX:+IgnoreUnrecognizedVMOptions -Dbloop.ignore-sig-int=true -cp /home/local/USERECE/.cache/coursier/v1/https/repo1.maven.org/maven2/io/github/alexarchambault/bleep/bloop-frontend_2.12/1.5.11-sc-3/bloop-frontend_2.12-1.5.11-sc-3.jar/home/local/USERECE//USERECE/.cache/coursier/v1/https/repo1.maven.org/maven2/ch/epfl/scala/bloop-config_2.12/1.5.5/bloop-config_2.12-1.5.5.jar:/home/local/USERECE/.cache/coursier/v1/https/repo1.maven.org/maven2/com/github/alexarchambault/case-app_2.12/2.0.6/case-app_2.12-2.0.6.jar:/home/local/USERECE/.cache/coursier/v1/https/repo1.maven.org/maven2/io/github/alexarchambault/libdaemo

what is the above process
why the above process take ram when debugging its suddenly take 1.5gb

**Comment by tgodzik** [comment]
This is the compile server that Scala CLI uses to compile the code.

**Reply by ECEAthipanbose** [reply]
okay but its get more ram size for simple scala program , so its cause slow the ide ,there is any other way to reduce thier process ram size?

**Reply by tgodzik** [reply]
So for Scala CLI you would need to set BLOOP_JAVA_OPTS and make sure that scala-cli's bloop is restarted via `scala-cli --power bloop exit` and then any other command.

Here are the defaults https://github.com/scala-cli/bloop-core/blob/0d0937ecff405f214586bede5f5f1f3b44399e7a/bloop-rifle/src/bloop/rifle/BloopRifleConfig.scala#L76

I would add something like `-Xmx1g`

**Reply by ECEAthipanbose** [reply]
still ram size near 1.5gb
there are thre process run in ide 
one is scala clie used ffor compile code 
what is the remaining two and thier purpose ?

    java -Xss4m -Xms100m -Xmx1G -classpath /home/local/USERECE/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalameta/metals_2.13/1.2.2/metals_2.13-1.2.2.jar:/home/local/USERECE/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.12/scala-library-2.13.12.jar:.... its take near 1gb
    
    java -cp /home/local/USERECE/.cache/coursier/v1/https/repo1.maven.org/maven2/org/virtuslab/scala-cli/cli_3/1.1.2/cli_3-1.1.2.jar:/home/local/USERECE/.cache/coursier/v1/https/repo1.maven.org/maven2/com/github/alexarchambault/case-app_3/2.1.0-M26/case-app_3-2.1.0-M26.jar:/... its take near 300mb
    
    scalacli takes near 1gb totally greater than 2gb of ram size
    
    how to reduce thier ram size

**Reply by tgodzik** [reply]
One is Bloop that is used for compilation, then we have scala CLI itself, which should not take up too much memory especially if we use the native image version (unless native image starts a separate JVM server) and the third is Metals itself. I don't think we can go much lower since JVM is usually using a bit more memory and both Metals and Bloop need to compile things, which means they need to use that memory for all the things the compiler would.
